ambry-replication/src/main/java/com.github.ambry.replication/ReplicaThread.java

```diff
@@ -518,13 +514,9 @@ private void checkNeedWaitTime(ReplicaMetadataResponseInfo replicaMetadataRespon
       // from the client. This is done only when the replication lag with that node is less than
       // replicationMaxLagForWaitTimeInBytes
       Thread.sleep(replicationConfig.replicaWaitTimeBetweenReplicasMs);
-      needToWaitForReplicaLag = false;
-    }
-    if (remoteColo) {
-      replicationMetrics.interColoReplicationWaitTime.update(SystemTime.getInstance().milliseconds() - startTime);
-    } else {
-      replicationMetrics.intraColoReplicationWaitTime.update(SystemTime.getInstance().milliseconds() - startTime);
+      waitEnabled = false;
```

Comment:
We just wait only once for a node. So, if we have waited for any replica in that node, we don't need to wait for other replicas